nengo_loihi/hardware/allocators.py

```diff
@@ -214,3 +214,202 @@ def get_chip(i):
+                assert block.compartment.n_compartments == ens.n_neurons
+                block_rates[block] = rates
+            else:
+                block_rates[block] = rates[comp_idxs]
+
+    return block_rates
+
+
+def compute_block_conns(block_map, block_rates=None, conns_in=False):  # noqa: C901
+    # --- store number of axons from block i to block j
+    block_conns = {k: {} for k in block_map}
+    if conns_in:
+        block_conns_in = {k: {} for k in block_map}
+
+    synapse_block_map = {}
+    for i, block_i in block_map.items():
+        for synapse in block_i.synapses:
+            assert id(synapse) not in synapse_block_map
+            synapse_block_map[id(synapse)] = i
+
+    for i, block_i in block_map.items():
+        for axon in block_i.axons:
+            j = synapse_block_map[id(axon.target)]
+
+            if i == j:
+                # don't care about self connections
+                continue
+
+            # use non-zero value as default, so that even if all rates are zero, this
+            # still gets recognized as a connection from i to j
+            block_conns[i].setdefault(j, 1e-16)
+            if conns_in:
+                block_conns_in[j].setdefault(i, 1e-16)
+
+            if block_rates is None:
+                val = axon.n_axons
+            elif block_i not in block_rates:
+                raise KeyError(f"block {block_i} not in block_rates")
+            else:
+                rates = block_rates[block_i]
+                comp_idxs = np.arange(block_i.compartment.n_compartments)
+                axon_ids = axon.map_axon(comp_idxs)
+                assert axon_ids.size == rates.size
+                val = rates[axon_ids >= 0].sum()
+
+            block_conns[i][j] += val
+            if conns_in:
+                block_conns_in[j][i] += val
+
+    return (block_conns, block_conns_in) if conns_in else block_conns
+
+
+def measure_interchip_conns(board, block_rates=None):
+    i = 0
+    block_map = {}
+    block_chip = {}
+    for chip in board.chips:
+        chip_idx = board.chip_idxs[chip]
+        for core in chip.cores:
+            # core_idx = chip.core_idxs[core]
+            for block in core.blocks:
+                block_map[i] = block
+                block_chip[i] = chip_idx
+                i += 1
+
+    block_conns = compute_block_conns(block_map, block_rates=block_rates)
+
+    stats = {"interchip": 0, "intrachip": 0}
+    stats["interchip_pairs"] = []
+    stats["intrachip_pairs"] = []
+    for i, block in block_map.items():
+        chip_idx_i = block_chip[i]
+        for j, weight in block_conns[i].items():
+            assert i != j, "compute_block_conns should skip recurrent connections"
+            chip_idx_j = block_chip[j]
+            key = "intrachip" if chip_idx_i == chip_idx_j else "interchip"
+            stats[key] += weight
+            stats[f"{key}_pairs"].append((block_map[i], block_map[j]))
+
+    return stats
+
+
+class GreedyComms(Greedy):
```

Comment:
Honestly, I would probably go with `Interchip` rather than `Intrachip`. In the `Greedy` case, you're right that what it's being greedy about is intra-chip communication, in that it puts high-connectivity connections on the same chip greedily. But in both cases, what they're trying to optimize is inter-chip communication, so naming them `Interchip` highlights their purpose. Further to that point, `Interchip` connotes that they're for networks that are multi-chip; to me, `GreedyIntrachip` sounds kind of like it's only for networks that are all within one chip. Also, `PartitionIntrachip` doesn't really make sense to me; it sounds like it's somehow partitioning things within the chip, which it's not (and while we could use intra for one and inter for the other, that just seems awkward). Finally, the `inter` prefix is just a bit more common and accessible than `intra`, and IMO both comes to mind and flows off the tongue a little more readily.

Reply:
Yeah, you're right that interchip is probably easier to remember. I'll switch!